datafusion-postgres/src/handlers.rs

```diff
@@ -176,36 +485,95 @@ impl ExtendedQueryHandler for DfSessionService {
+            let dialect = GenericDialect {};
+            let stmts = SqlParser::parse_sql(&dialect, &stmt_string)
+                .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
+            if let Statement::ShowVariable { variable } = &stmts[0] {
+                let resp = self.handle_show(variable).await?;
+                return Ok(Response::Query(resp));
+            }
+        }
 
+        // Otherwise, treat it as a normal prepared statement.
+        let plan = &portal.statement.statement;
         let param_types = plan
             .get_parameter_types()
             .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
-
         let param_values =
             datatypes::deserialize_parameters(portal, &ordered_param_types(&param_types))?;
-
         let plan = plan
             .clone()
             .replace_params_with_values(&param_values)
             .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
 
-        let dataframe = self
-            .session_context
+        let sc_guard = self.session_context.read().await;
+        let dataframe = sc_guard
             .execute_logical_plan(plan)
             .await
             .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
 
-        let resp = datatypes::encode_dataframe(dataframe, &portal.result_column_format).await?;
+        let resp =
+            datatypes::encode_dataframe(dataframe, &portal.result_column_format).await?;
         Ok(Response::Query(resp))
     }
 }
 
 fn ordered_param_types(types: &HashMap<String, Option<DataType>>) -> Vec<Option<&DataType>> {
-    // Datafusion stores the parameters as a map.  In our case, the keys will be
-    // `$1`, `$2` etc.  The values will be the parameter types.
+    // Datafusion stores the parameters as a map. In our case, the keys will be
+    // `$1`, `$2` etc. The values will be the parameter types.
+    let mut types_vec = types.iter().collect::<Vec<_>>();
+    types_vec.sort_by(|a, b| a.0.cmp(b.0));
+    types_vec.into_iter().map(|pt| pt.1.as_ref()).collect()
+}
 
-    let mut types = types.iter().collect::<Vec<_>>();
-    types.sort_by(|a, b| a.0.cmp(b.0));
-    types.into_iter().map(|pt| pt.1.as_ref()).collect()
+/// Register a UDF called `current_schemas` that takes a boolean and returns an array containing "public".
+fn register_current_schemas_udf(ctx: &mut SessionContext) -> datafusion::error::Result<()> {
+    let current_schemas_fn = Arc::new(move |args: &[ColumnarValue]| -> datafusion::error::Result<ColumnarValue> {
+        // We ignore the input value; just return a constant list containing "public".
+        let num_rows = match &args[0] {
+            ColumnarValue::Array(array) => array.len(),
+            ColumnarValue::Scalar(_) => 1,
+        };
+        // Build a ListArray containing "public"
+        let mut list_builder = ListBuilder::new(StringBuilder::new());
+        for _ in 0..num_rows {
+            list_builder.values().append_value("public");
```

Comment:
Can we avoid hardcode public here? Is it possible to read schemas from `SessionContext`?